File: backend/scm/src/main/java/com/scm/scm/events/vao/Event.java
package com.scm.scm.events.vao;


import lombok.AllArgsConstructor;
import lombok.Data;
import org.springframework.data.annotation.Id;

import java.security.SecureRandom;
import java.time.LocalDateTime;

@Data
@AllArgsConstructor
public class Event {

    @Id
    private String id;
    private String user;
    private String contact;
    private EventState eventState;
    private String propKey;
    private String prevState;
    private String currentState;
    private LocalDateTime eventTime;

    public Event() {
    }

    public Event(String user, String contact, EventState eventState) {
        this.user = user;
        this.contact = contact;
        this.eventState = eventState;
        this.propKey = "";
        this.prevState = "";
        this.currentState = "";
    }

    private static final SecureRandom random = new SecureRandom();
    public String generateId() {
        return System.nanoTime() + "-" + random.nextInt(10_000);
    }

}

File: frontend/scm/src/models/Event.ts
export enum EventState {
    CREATED = 'CREATED',
    UPDATED = 'UPDATED',
    DELETED = 'DELETED',
    TAG_ADD = 'TAG_ADD',
    TAG_REMOVED = 'TAG_REMOVED',
    PROP_ADD = 'PROP_ADD',
    PROP_REMOVED = 'PROP_REMOVED',
    REVERTED = 'REVERTED',
}

export interface Event {
    id: string;
    user: string;
    contact: string;
    eventState: EventState;
    propKey: string;
    prevState: string;
    currentState: string;
    eventTime: Date;
}